packages/graphiql/src/components/__tests__/helpers/codeMirror.js

```diff
@@ -0,0 +1,81 @@
+export function MockCodeMirror(node, { value, ...options }) {
+  let _eventListeners = {};
+  const mockTextArea = document.createElement('textarea');
```

Comment:
I don't think it matters now but it may eventually, but our instantiation pattern for GraphiQL query editor is a bit different from the generic Codemirror example that tends to use`<textarea />`.

We actually end up using a `<section />`:

[Codemirror Invocation](https://github.com/graphql/graphiql/blob/master/packages/graphiql/src/components/QueryEditor.js#L78)
[`this_node` attached via old ref pattern from `<section />` element](https://github.com/graphql/graphiql/blob/master/packages/graphiql/src/components/QueryEditor.js#L219)

Reply:
we target a `<section />` to instantiate CodeMirror but code mirror renders some more elements including a `<textarea />` inside that root node:

<img width="780" alt="image" src="https://user-images.githubusercontent.com/26845525/71483562-70f7e400-27d6-11ea-84de-9ccd22bb1a51.png">

they handle the typing and presentation in a pretty complex way it seems, where they use the type area as a temporary moving cursor and then render out typed text in some presentational elements. i figured for our testing purposes it'd just be easier to use a base text area and the abstracted mocked internals of CodeMirror for our testing purposes and keep the API our code interacts with the same. If you have a better idea let me know though.

Reply:
indeed yes! just pointing out there will be a subtle difference in the DOM hierarchy, but I think the only delta would be caused by container targets for full-height container

Reply:
either way this is really great, and we can use this mock for codemirror-graphql I hope, which we still need to convert from mocha to jest

Reply:
gotcha, yeah this might need some changes as we go. I already added a few more elements to the mock CodeMirror DOM in another PR that uses this branch as its base because it need to listen for click drag events on the gutter bar.